File: Assignment 4/Python/01. time taken .py

class time:
    def __init__ (self,h,m,s):
        self.h = h
        self.m = m
        self.s = s

def diff(start,end):
    while end.s<start.s:
        end.s+=60
        end.m-=1
    while end.m<start.m:
        end.m+=60
        end.h-=1
    
    diff_h = end.h-start.h
    diff_m = end.m-start.m
    diff_s = end.s-start.s

    ans = diff_h * 3600 + diff_m * 60 + diff_s
    return ans


start = time(int(input("h1: ")),
             int(input("m1: ")),
             int(input("s1: ")))
end = time(int(input("h2: ")),
             int(input("m2: ")),
             int(input("s2: ")))

ans = diff(start,end)
print("ans = ",ans)



# OUTPUT:
# h1: 3
# m1: 4
# s1: 5
# h2: 6
# m2: 7
# s2: 8
# ans =  10983 

File: Assignment 4/C/01. time taken.c
#include <stdio.h>

typedef struct time {
    int h;
    int m;
    int s;
}time;

int diff(time start,time end){
    while(end.s<start.s){
        end.s+=60;
        end.m--;
    }
    while(end.m<start.m){
        end.m+=60;
        end.h--;
    }
    time diff;
    diff.h = end.h-start.h;
    diff.m = end.m-start.m;
    diff.s = end.s-start.s;
    
    int ans = (diff.h*3600)+(diff.m*60)+diff.s;
    return ans;
}

int main(){
    time start;
    time end;
    printf("enter start time :\n");
    printf("h: ");
    scanf("%d",&start.h);
    printf("m: ");
    scanf("%d",&start.m);
    printf("s: ");
    scanf("%d",&start.s);

    printf("enter end time :\n");
    printf("h: ");
    scanf("%d",&end.h);
    printf("m: ");
    scanf("%d",&end.m);
    printf("s: ");
    scanf("%d",&end.s);

    int dif = diff(start,end);
    printf("diff = %d seconds",dif);

    return 0;
}



//OUTPUT:

// enter start time :
// h: 2
// m: 3
// s: 4
// enter end time :
// h: 2
// m: 34
// s: 56
// diff = 1912 seconds
